velox/common/memory/MemoryAllocator.h

```diff
@@ -311,6 +311,47 @@ class MemoryAllocator : public std::enable_shared_from_this<MemoryAllocator> {
   /// multiple of 'alignmentBytes'.
   static void alignmentCheck(uint64_t allocateBytes, uint16_t alignmentBytes);
 
+  /// Causes 'failure' to occur in memory allocation calls. This is a test-only
+  /// function for validating otherwise unreachable error paths. If 'persistent'
+  /// is false, then we only inject failure once in the next call. Otherwise, we
+  /// keep injecting failures until next 'testingClearFailureInjection' call.
+  enum class Failure {
+    kNone,
+    /// Mimic case of not finding anything to advise away.
+    ///
+    /// NOTE: this only applies for MmapAllocator.
+    kMadvise,
+    /// Mimic running out of mmaps for process.
+    ///
+    /// NOTE: this only applies for MmapAllocator.
+    kMmap,
+    /// Mimic the actual memory allocation failure.
+    kAllocate,
+    /// Mimic the case that exceeds the memory allocator's internal cap limit.
+    ///
+    /// NOTE: this only applies for MmapAllocator.
+    kCap
+  };
+  void testingSetFailureInjection(Failure failure, bool persistent = false) {
+    injectedFailure_ = failure;
+    isPersistentFailureInjection_ = persistent;
+  }
+
+  void testingClearFailureInjection() {
```

Comment:
Have we updated the coding style guidelines to document the naming for the test-only methods?

Reply:
We haven't yet: D38372050.